### DISCUSSION TITLE
Keep X number of old images - Retention

### DISCUSSION THREAD

**Discussion by jgdub** [containrrr/watchtower#1403] [Q&A]
Is it possible to configure watchtower cleanup so that it will keep a specific number of old images during cleanup?

I would like to have an automatic clean up but I would like to also keep the last 2 or 3.

Thanks

**Comment by piksel** [comment]
No, the `--cleanup` is all or nothing right now. It could be implemented by querying for images using the same tag and sorting them by creation date. It should probably be done through scripting together with #1231, though.

**Reply by simonaronsson** [reply]
We've discussed this numerous times and unfortunately that won't solve the issue either. Once a new image is pulled for the tag, the old one will be untagged by docker.
